ImageProcessing.cpp

```diff
@@ -0,0 +1,55 @@
+#include "ImageProcessing.h"
+
+
+void ImageProcessing::processImages(std::string &imageDir) {
+    int moleCounter = 0;
+    double paddingFactor = 3.0;
+
+    //    Load the Image and Preprocess
+    cv::Mat image = cv::imread("skin_image.png");
+    cv::Mat gray;
+    cv::cvtColor(image, gray, cv::COLOR_BGR2GRAY);
+    cv::GaussianBlur(gray, gray, cv::Size(5, 5), 0);
+
+    //    Segmentation using Adaptive Thresholding
+    cv::Mat thresh;
+    cv::adaptiveThreshold(gray, thresh, 255, cv::ADAPTIVE_THRESH_MEAN_C, cv::THRESH_BINARY_INV, 11, 2);
```

Comment:
Document why did you choose these numbers? 255, 11, 2.
Put yourself in the shoes of someone who comes in half a year to fix a bug here...